File: src/fractal/interpret/observed.py


#####
##### Statistics for games with ternary rewards
#####

@dataclass
class TernaryOutcomeStatistics:
  num_won: Int
  num_draw: Int
  num_lost: Int

function TernaryOutcomeStatistics(rewards::AbstractVector{<:Number})
  num_won  = count(==(1), rewards)
  num_draw = count(==(0), rewards)
  num_lost = count(==(-1), rewards)
  @assert num_won + num_draw + num_lost == length(rewards)
  return TernaryOutcomeStatistics(num_won, num_draw, num_lost)
end

function TernaryOutcomeStatistics(report::Report.Evaluation)
  return TernaryOutcomeStatistics(report.rewards)
end

File: src/fractal/interpret/observed.rs
//! Statistics for games with ternary rewards

pub struct TernaryOutcomeStatistics {
    n_won: i16,
    n_draw: i16,
    n_lost: i16,
}

impl TernaryOutcomeStatistics {
    fn from_rewards(rewards::AbstractVector{<:Number}) {
        num_won  = count(==(1), rewards);
        num_draw = count(==(0), rewards);
        num_lost = count(==(-1), rewards);
        // @assert num_won + num_draw + num_lost == length(rewards)
        TernaryOutcomeStatistics {n_won, n_draw, n_lost }
    }

    fn from_evaluated(evaluated: &Evaluated) {
        TernaryOutcomeStatistics { report.rewards }
    }
}